source/agora/app.d

```diff
@@ -35,20 +36,37 @@ version (unittest) void main () { } else:
 private int main (string[] args)
 {
     CommandLine cmdln;
-    auto help = parseCommandLine(cmdln, args);
-    if (help.helpWanted)
+
+    try
     {
-        defaultGetoptPrinter("The Agora node", help.options);
-        return 0;
+        auto help = parseCommandLine(cmdln, args);
+        if (help.helpWanted)
+        {
+            defaultGetoptPrinter("The Agora node", help.options);
+            return 0;
+        }
     }
-
-    auto config = parseConfigFile(cmdln);
-    if (cmdln.config_check)
+    catch (Exception ex)
     {
-        writefln("Config file '%s' succesfully parsed.", cmdln.config_path);
-        return 0;
+        writefln("Error parsing command-line arguments '%s': %s", args.join(" "), ex.msg);
```

Comment:
Use `%(%s %)` and you don't need the `join`

Reply:
Also `ex.message()`

Reply:
ex.message? I thought this was only an Ocean thing?

Reply:
> Use %(%s %) and you don't need the join

Nope:

```
Error parsing command-line arguments ''U' 'n' 'r' 'e' 'c' 'o' 'g' 'n' 'i' 'z' 'e' 'd' ' ' 'o' 'p' 't' 'i' 'o' 'n' ' ' '-' 'x'': %
```

Reply:
Not on the message, just on `args`

Reply:
Yeah I'm not sure why it went bad the first time I tried it. It looks ok now, thanks.

Reply:
> ex.message? I thought this was only an Ocean thing?

No we added it to `druntime`